src/sst/core/model/python/pymodel_unitalgebra.cc

```diff
@@ -0,0 +1,389 @@
+    PyObject* ret = PyObject_CallObject((PyObject*)&PyModel_UnitAlgebraType, argList);    
+    Py_DECREF(argList);
+
+    UnitAlgebraPy_t *ret_ua = (UnitAlgebraPy_t*)ret;
+    UnitAlgebraPy_t *other_ua = (UnitAlgebraPy_t*)other;
+    *(ret_ua->obj) *= *(other_ua->obj);
+    return ret;
+}
+    
+static PyObject* unitAlgebraDivIP(PyObject *self, PyObject *other)
+{
+    UnitAlgebraPy_t *self_ua = (UnitAlgebraPy_t*)self;
+    UnitAlgebraPy_t *other_ua = (UnitAlgebraPy_t*)other;
+    *(self_ua->obj) /= *(other_ua->obj);
+
+    PyObject *argList = Py_BuildValue("(O)", self);
+    PyObject* ret = PyObject_CallObject((PyObject*)&PyModel_UnitAlgebraType, argList);    
+    Py_DECREF(argList);
+    
+    return ret;
+}
+
+static PyObject* unitAlgebraDiv (PyObject *self, PyObject *other)
+{
+    PyObject *argList = Py_BuildValue("(O)", self);
+    PyObject* ret = PyObject_CallObject((PyObject*)&PyModel_UnitAlgebraType, argList);    
+    Py_DECREF(argList);
+
+    UnitAlgebraPy_t *ret_ua = (UnitAlgebraPy_t*)ret;
+    UnitAlgebraPy_t *other_ua = (UnitAlgebraPy_t*)other;
+    *(ret_ua->obj) /= *(other_ua->obj);
+    return ret;
+}
+
+static PyObject* unitAlgebraToLong(PyObject *self)
+{
+    UnitAlgebraPy_t *self_ua = (UnitAlgebraPy_t*)self;
+    int64_t val = self_ua->obj->getRoundedValue();
+    PyObject* ret = PyLong_FromLong(val);
+    return ret;
+}
+
+static PyObject* unitAlgebraNegate(PyObject *self)
+{
+    PyObject *argList = Py_BuildValue("(O)", self);
+    PyObject* ret = PyObject_CallObject((PyObject*)&PyModel_UnitAlgebraType, argList);    
+    Py_DECREF(argList);
+    UnitAlgebraPy_t *ret_ua = (UnitAlgebraPy_t*)ret;
+    std::cout << ret_ua->obj->toString() << std::endl;
+    *(ret_ua->obj) *= -1;
+    std::cout << ret_ua->obj->toString() << std::endl;
+    return ret;
+}
+
+
+
+PyNumberMethods PyModel_UnitAlgebraNumMeth = {
+    (binaryfunc)unitAlgebraAdd,    // binaryfunc nb_add
+    (binaryfunc)unitAlgebraSub,    // binaryfunc nb_subtract
+    (binaryfunc)unitAlgebraMul,    // binaryfunc nb_multiply
+    SST_NB_DIVIDE((binaryfunc)unitAlgebraDiv)    // binaryfunc nb_divide, py2 only
+    nullptr,                // binaryfunc nb_remainder
+    nullptr,                // binaryfunc nb_divmod
+    nullptr,                // ternaryfunc nb_power
+    (unaryfunc)unitAlgebraNegate,                // unaryfunc nb_negative
+    nullptr,                // unaryfunc nb_positive
+    nullptr,                // unaryfunc nb_absolute
+    nullptr,                // inquiry nb_nonzero (py2) nb_bool (py3)       /* Used by PyObject_IsTrue */
+    nullptr,                // unaryfunc nb_invert
+    nullptr,                // binaryfunc nb_lshift
+    nullptr,                // binaryfunc nb_rshift
+    nullptr,                // binaryfunc nb_and
+    nullptr,                // binaryfunc nb_xor
+    nullptr,                // binaryfunc nb_or
+    SST_NB_COERSE                // coercion nb_coerce Python 2 only
```

Comment:
only necessary to appease my spelling tyranny, but COERSE -> COERCE

Reply:
Fixed.